src/main/java/com/znsio/teswiz/runner/BrowserDriverManager.java

```diff
@@ -195,25 +209,165 @@ private static DriverManagerType setupBrowserDriver(TestExecutionContext testExe
+
+        String imageName = "";
+        String containerId = "";
+
+        try {
+            Yaml yaml = new Yaml();
+            FileInputStream fis = new FileInputStream("docker-compose.yml");
+            Map<String, Object> data = yaml.load(fis);
+            Map<String, Object> services = (Map<String, Object>) data.get("services");
+            Map<String, Object> selenium = (Map<String, Object>) services.get("chrome");
+            imageName = (String) selenium.get("image");
+            LOGGER.info("Selenium image: " + imageName);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+
+        List<Container> containers = dockerClient.listContainersCmd()
+                .withShowAll(true)
+                .exec();
+        for (Container container : containers) {
+            if (container.getImage().equals(imageName)) {
+                containerId = container.getId();
+            }
+        }
+
+        assert containerId != null;
+        ExecCreateCmdResponse execCreateCmdResponse = dockerClient.execCreateCmd(containerId)
+                .withCmd("sh", "-c", FETCH_CONTAINER_BROWSER_VERSION_COMMAND)
+                .withAttachStdout(true)
+                .withAttachStderr(true)
+                .exec();
+
+        String browserVersion = executeAndGetBrowserVersion(dockerClient, execCreateCmdResponse.getId());
+        LOGGER.info("Browser Version in container: " + browserVersion);
+        return browserVersion;
+    }
+
+    private static String executeAndGetBrowserVersion(DockerClient dockerClient, String execId) {
+        StringBuilder logOutput = new StringBuilder();
+        Pattern versionPattern = Pattern.compile("Google Chrome (\\d+\\.\\d+\\.\\d+\\.\\d+)");
+        try {
+            dockerClient.execStartCmd(execId).exec(new ExecStartResultCallback() {
+                @Override
+                public void onNext(Frame frame) {
+                    logOutput.append(frame.toString());
+                }
+            }).awaitCompletion();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        Matcher matcher = versionPattern.matcher(logOutput.toString());
+        if (matcher.find()) {
+            return matcher.group(1);
+        }
+        return null;
+    }
+
+    private static String getLocalBrowserVersionFor(String browserType) {
+        String binaryPath = "";
+        String os = System.getProperty("os.name").toLowerCase();
+        switch (browserType) {
+            case "chrome":
+                binaryPath = getChromeBinaryPath(os);
+                break;
+            case "firefox":
+                binaryPath = getFirefoxBinaryPath(os);
+                break;
+            case "safari":
+                return getSafariVersion();
+            default:
+                throw new InvalidTestDataException("Invalid browser : " + browserType);
+        }
+        ChromeOptions chromeOptions = new ChromeOptions();
```

Comment:
Is this required we have helper methods to get binary paths.

Reply:
Chrome options was used to set the binary path, removed and updated as we have helper methods to get the path